src/Components/OATGraphViewer/Internal/Utils.tsx

```diff
@@ -1,171 +1,178 @@
-                (element.data as DtdlRelationship).name ===
-                    `${OATRelationshipHandleName}_${relationshipIndex}`
-        )
-    ) {
-        relationshipIndex++;
-    }
-    return relationshipIndex;
-};
+export const CONTEXT_CLASS_BASE = 'dtmi:dtdl:context;2';
+export const DEFAULT_NODE_POSITION = 25;
 
-const getNextComponentIndex = (
-    sourceId: string,
-    targetName: string,
-    elements: (ElementNode | ElementEdge)[]
-) => {
-    let componentIndex = 0;
-    const match = elements.find((element) => element.id === sourceId);
-    if (match) {
-        const int = match.data as DtdlInterface;
-        while (
-            int &&
-            int.contents.some(
-                (content) =>
-                    content['schema'] &&
-                    content.name === `${targetName}_${componentIndex}`
-            )
-        ) {
-            componentIndex++;
-        }
-    }
-    return componentIndex;
-};
+//#region Add relationships
 
 export const addTargetedRelationship = (
     sourceId: string,
-    relationship: DtdlInterfaceContent,
+    relationship: DtdlRelationship,
     elements: (ElementNode | ElementEdge)[]
 ) => {
-    const nextRelIndex = getNextRelationshipIndex(sourceId, elements);
-    const name =
-        relationship.name || `${OATRelationshipHandleName}_${nextRelIndex}`;
+    // logDebugConsole(
+    //     'debug',
+    //     '[START] addTargetedRelationship. {source, relationship, elements}',
+    //     sourceId,
+    //     relationship,
+    //     elements
+    // );
+    const name = relationship.name;
     const id = relationship['@id'] || `${sourceId}_${name}`;
     const relationshipEdge = new ElementEdge(
         id,
         '',
-        OATRelationshipHandleName,
+        OAT_GRAPH_REFERENCE_TYPE,
         '',
         sourceId,
-        OATRelationshipHandleName,
+        OAT_RELATIONSHIP_HANDLE_NAME,
         relationship['target'],
-        OATRelationshipHandleName,
+        OAT_RELATIONSHIP_HANDLE_NAME,
         {
-            ...relationship,
-            '@type': OATRelationshipHandleName,
+            '@type': DTDLType.Relationship,
+            '@id': relationship['@id'],
             name
         }
     );
 
     elements.push(relationshipEdge);
+    // logDebugConsole(
+    //     'debug',
+    //     '[END] addTargetedRelationship. {elements}',
+    //     elements
+    // );
     return relationshipEdge;
 };
 
 export const addUntargetedRelationship = (
-    sourceId: string,
-    relationship: DtdlInterfaceContent,
+    sourceModel: DtdlInterface,
+    relationship: DtdlRelationship,
     modelPositions: IOATModelPosition[],
     elements: (ElementNode | ElementEdge)[]
 ) => {
-    const nextRelIndex = getNextRelationshipIndex(sourceId, elements);
-    const name =
-        relationship.name || `${OATRelationshipHandleName}_${nextRelIndex}`;
-    const id = relationship['@id'] || `${sourceId}_${name}`;
+    logDebugConsole(
```

Comment:
Other calls to `logDebugConsole` in this file are commented out. I'm curious if this set should be as well.